File: java/IEC61970/Base/Wires/RegulatingCondEq.java



/**
 * A type of conducting equipment that can regulate a quantity (i.e. voltage or
 * flow) at a specific point in the network.
 * @author pmora
 * @updated 15-Dec-2023 1:39:42 PM
 */
public class RegulatingCondEq extends EnergyConnection {

	/**
	 * Specifies the regulation status of the equipment.  True is regulating, false is
	 * not regulating.
	 */
	private boolean controlEnabled;
	/**
	 * The regulating control scheme in which this equipment participates.
	 */
	private RegulatingControl RegulatingControl;

	public RegulatingCondEq(){

	}

	public void finalize() throws Throwable {
		super.finalize();
	}

	public boolean getcontrolEnabled(){
		return controlEnabled;
	}

	public RegulatingControl getRegulatingControl(){
		return RegulatingControl;
	}

	/**
	 * 
	 * @param newVal
	 */
	public void setcontrolEnabled(boolean newVal){
		controlEnabled = newVal;
	}

	/**
	 * 
	 * @param newVal
	 */
	public void setRegulatingControl(RegulatingControl newVal){
		RegulatingControl = newVal;
	}

}

File: py/IEC61970/Base/Wires/RegulatingCondEq.py
# Converted by an OPENAI API call using model: gpt-3.5-turbo-1106
from CIM_STD_PYTHON.TC57CIM.IEC61970.Base.Wires.EnergyConnection import EnergyConnection
from CIM_STD_PYTHON.TC57CIM.IEC61970.Base.Wires.RegulatingControl import RegulatingControl


class RegulatingCondEq(EnergyConnection):

    def __init__(self):
        super().__init__()
        self.control_enabled = False
        self.regulating_control = RegulatingControl()

    def get_control_enabled(self) -> bool:
        return self.control_enabled

    def get_regulating_control(self) -> RegulatingControl:
        return self.regulating_control

    def set_control_enabled(self, new_val: bool):
        self.control_enabled = new_val

    def set_regulating_control(self, new_val: RegulatingControl):
        self.regulating_control = new_val

